exporter/util.go

```diff
@@ -149,21 +150,42 @@ func (ic *importContext) emitNotebookOrRepo(path string) {
 func (ic *importContext) getAllDirectories() []workspace.ObjectStatus {
 	if len(ic.allDirectories) == 0 {
 		objects := ic.getAllWorkspaceObjects()
-		for _, v := range objects {
-			if v.ObjectType == workspace.Directory {
-				ic.allDirectories = append(ic.allDirectories, v)
+		ic.wsObjectsMutex.Lock()
+		if len(ic.allDirectories) == 0 {
+			for _, v := range objects {
+				if v.ObjectType == workspace.Directory {
+					ic.allDirectories = append(ic.allDirectories, v)
+				}
 			}
 		}
+		ic.wsObjectsMutex.Unlock()
```

Comment:
In general, I think it's preferable to do `defer ic.wsObjectsMutex.Unlock()` earlier up, in case of an early return or panic.